v3/metadata/metadata.go

```diff
@@ -70,3 +92,20 @@ func httpGet(ctx context.Context, url string) (string, error) {
 
 	return string(body), nil
 }
+
+func getFileMetaDataValue(file *os.File, endpoint string) (string, error) {
+	scanner := bufio.NewScanner(file)
+	for scanner.Scan() {
```

Comment:
It feels weird to read the raw disk & search for a string inside of it like that

Did you consider using `Disk.GetFilesystem` then `FileSystem.OpenFile` ?